src/main/java/bwapi/Client.java

```diff
@@ -258,67 +250,17 @@ void update(final EventHandler handler) {
             }
             catch (Exception e) {
                 System.err.println("failed, disconnecting");
-                if (debugConnection) {
+                if (configuration.debugConnection) {
                     e.printStackTrace();
                 }
                 disconnect();
                 return;
             }
         }
-        for (int i = 0; i < gameData.getEventCount(); i++) {
-            handler.operation(gameData.getEvents(i));
-        }
-    }
-
-    String eventString(final int s) {
-        return gameData.getEventStrings(s);
-    }
```

Comment:
The logic for applying changes to the game has moved to Game and SideEffect/SideEffectQueue